File: release/fixtures/native-resource/app/src/main/java/com/tongsr/kaleido/matrix/nativeprobe/MainActivity.java
package com.tongsr.kaleido.matrix.nativeprobe;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

public final class MainActivity extends Activity {
    static {
        System.loadLibrary("kaleido_probe");
    }

    private static native int nativeAnswer();

    @Override
    protected void onCreate(Bundle state) {
        super.onCreate(state);
        var resource = getString(R.string.probe_value);
        var answer = nativeAnswer();
        if (!resource.equals("resource-ok") || answer != 42) {
            throw new IllegalStateException("runtime probe failed");
        }
        Log.i("KaleidoRuntimeProbe",
                "KALEIDO_PROBE_PASS resource=" + resource + " native=" + answer);
    }
}

File: release/fixtures/native-resource/app/src/main/kotlin/com/tongsr/kaleido/matrix/nativeprobe/MainActivity.kt
package com.tongsr.kaleido.matrix.nativeprobe

import android.app.Activity
import android.os.Bundle
import android.util.Log

class MainActivity : Activity() {
    override fun onCreate(state: Bundle?) {
        super.onCreate(state)
        val resource = getString(R.string.probe_value)
        val answer = nativeAnswer()
        if (resource != "resource-ok" || answer != 42) {
            throw IllegalStateException("runtime probe failed")
        }
        Log.i(
            "KaleidoRuntimeProbe",
            "KALEIDO_PROBE_PASS resource=$resource native=$answer",
        )
    }

    companion object {
        init {
            System.loadLibrary("kaleido_probe")
        }

        @JvmStatic
        private external fun nativeAnswer(): Int
    }
}
